types/src/crypto/asymmetric_key.rs

```diff
@@ -743,3 +749,104 @@ mod detail {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    mod from_hex {
+        use super::*;
+        use rand::{rngs::StdRng, SeedableRng, RngCore};
+
+        #[test]
+        fn too_short() {
+            assert!(PublicKey::from_hex([0u8; 0]).is_err());
+            assert!(PublicKey::from_hex([0u8; 1]).is_err());
+        }
+
+        // The following atrocity is due to the crates `rand_core` and
+        // `k256::elliptic_curve::rand_core` not necessarily being the same.
+        mod myrng {
```

Comment:
Maybe use a [proptest][1] instead of this? We already have `crate::crypto::assymetric_key::public_key_arb()`...

Reply:
That's an awesome review and I learned a lot, thanks.
I think I should start over with a new PR.

Reply:
I backtracked this PR and added new commits from the start.